Fix missing symbol when printing guards

aorenste opened the pull request:
Fixes #165177 

When converting guards to sources if we were unable to get the expected symbol from symbol_to_source then try to get it from var_to_sources.

I was unable to make a simpler repro than what was described in the issue (which relies on llama3 - so inappropriate for a unit test).

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #165723



cc @ezyang @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

aorenste committed:
WIP: testing fix for #165177

[ghstack-poisoned]

aorenste committed:
Update on "WIP: testing fix for #165177"




cc ezyang EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

ezyang commented:
What's the difference between symbol_to_source and var_to_source?

ezyang commented:
> we're swapping out the current FakeTensorMode but reusing the existing ShapeEnv

The reason this needs to be done is because when we do Dynamo tracing, fake tensors can have their metadata mutated. Thus, the fake tensor we allocated for any given tensor may no longer be valid for the beginning trace of the graph. Nor is it convenient to "clone" the input tensors before mutating them, since you have to preserve aliasing. So we just reconstruct the FakeTensorMode from scratch when we go to AOTAutograd. But the ShapeEnv must be preserved as Dynamo made decisions about what is dynamic or not / guards from the user code that is not in graph.

aorenste committed:
Update on "Fix missing symbol when printing guards"

Fixes #165177 

When converting guards to sources if we were unable to get the expected symbol from symbol_to_source then try to get it from var_to_sources.

I was unable to make a simpler repro than what was described in the issue (which relies on llama3 - so inappropriate for a unit test).




cc ezyang EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]
